src/main/java/com/metamx/common/lifecycle/Lifecycle.java

```diff
@@ -243,7 +250,10 @@ public void addMaybeStartHandler(Handler handler, Stage stage) throws Exception
   public void start() throws Exception
   {
     synchronized (handlers) {
-      started.set(true);
+      if (!started.compareAndSet(false, true)) {
```

Comment:
I feel like it makes sense to either not change the behavior, or to have it throw an error; silently skipping over double starts seems sketchy. At least the old behavior would probably blow up in some way and let us know something was being done wrong.